zerver/lib/user_groups.py

```diff
@@ -336,13 +336,36 @@ def check_user_group_name(group_name: str) -> str:
     return group_name
 
 
+def get_group_setting_value_for_api(
+    setting_value_group: UserGroup,
+) -> Union[int, AnonymousSettingGroupDict]:
+    if hasattr(setting_value_group, "named_user_group"):
+        return setting_value_group.id
+
+    return {
+        "direct_members": [member.id for member in setting_value_group.direct_members.all()],
+        "direct_subgroups": [
+            subgroup.id for subgroup in setting_value_group.direct_subgroups.all()
+        ],
```

Comment:
These queries are inefficient, use `.values_list(flat=True)` to only ask the database for the list of IDs.

Reply:
If we use `values_list(flat=True)`, then there would be no advantage of using `prefetch_related` and there would be a new query for each group, since `prefetch_related` caches the result for `group.direct_members.all()` call only.

Instead updated the `prefetch_related` call to set the queryset with `.only("id")`.